_includes/scripts.njk

```diff
@@ -40,3 +40,229 @@
 
   })();
 </script>
+
+<script>
+  (function(){
+    // WebComponent helpers
+    // setup attributes and the corresponding properties
+    function setupReflectedAttributes(klass) {
+      const schema = klass.attrs || {};
+      const descriptors = {};
+
+      for (const [name, type] of Object.entries(schema)) {
+        const prop = dashToCamel(name);  // attribute names are kebab, properties camel
+        descriptors[prop] = type === "bool"
+          ? reflectBool(name)
+          : reflectString(name);
+      }
+      Object.defineProperties(klass.prototype, descriptors);
+      klass.observedAttributes = Object.keys(schema);
+    }
+
+    function dashToCamel(attr) {
+      return attr.replace(/-([a-z])/g, (_, c) => c.toUpperCase());
+    }
+
+    function reflectString(attr) {
+      return {
+        get() {
+          return this.getAttribute(attr);
+        },
+        set(v) {
+          if (v == null) {
+            this.removeAttribute(attr);
+          } else {
+            this.setAttribute(attr, v);
+          }
+        }
+      };
+    }
+
+    function reflectBool(attr) {
+      return {
+        get() {
+          return this.hasAttribute(attr);
+        },
+        set(v) {
+          if (v) {
+            this.setAttribute(attr, "");
+          } else {
+            this.removeAttribute(attr);
+          }
+        }
+      };
+    }
+
+    class HumanDateElement extends HTMLElement {
+      static attrs = {
+        "abbreviate-months": "bool",
+        "always-months": "bool",
+        clickable: "bool",
```

Comment:
Currently the "reveal raw date" feature is not accessible via keyboard. Can we make these elements tab-focusable and respond to return/space keypresses? That means it should also probably have a button role and an indication of its purpose ("click to toggle between raw and readable date").

https://www.w3.org/WAI/ARIA/apg/patterns/button/

Reply:
Thanks, that’s awesome. I’m going to run over this with my screenreader etc. It’s my new hobby, listening to what websites sound like 😄